posts/2019-02-28-Rust-1.33.0.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: post
+title: "Announcing Rust 1.33.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.33.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.33.0 is as easy as:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.33.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1330-2019-02-28
+
+## What's in 1.33.0 stable
+
+The two largest features in this release are significant improvements to
+`const fn`s, and the stabilization of a new concept: "pinning."
+
+### `const fn` improvements
+
+With `const fn`, you can [now do way more
+things!](https://github.com/rust-lang/rust/pull/57175/) Specifically:
+
+* irrefutable destructuring patterns (e.g. `const fn foo((x, y): (u8, u8)) { ... }`)
+* `let` bindings (e.g. `let x = 1;`)
+* mutable `let` bindings (e.g. `let mut x = 1;`)
+* assignment (e.g. `x = y`) and assignment operator (e.g. `x += y`)
+  expressions, even where the assignment target is a projection (e.g. a struct
+  field or index operation like `x[3] = 42`)
+* expression statements (e.g. `3;`)
+
+You're also [able to call `const unsafe fn`s inside a `const
+fn`](https://github.com/rust-lang/rust/pull/57067/), like this:
+
+```rust
+const unsafe fn foo() -> i32 { 5 }
+const fn bar() -> i32 {
+    unsafe { foo() }
+}
+```
```

Comment:
It's not clear to me whether this is actually something we should call out as I believe `const unsafe fn` isn't actually interesting in that there's nothing unsafe that it can do yet?

cc @eddyb @oli-obk @rust-lang/compiler

Reply:
You can't e.g. dereference raw pointers, but it's certainly possible to call functions which have internal `unsafe` invariants (think `str::from_utf8_unchecked`).

Reply:
Ah, okay, that makes sense. Seems fine to include as such.